addon/components/flash-message.js

```diff
@@ -29,16 +29,16 @@ export default class FlashMessage extends Component {
   @bool('template')
   hasBlock;
 
-  @computed('flash.type', 'messageStyle')
+  @computed('messageStyle')
+  get _defaultMessageStylePrefix() {
+    const isFoundation = this.messageStyle === 'foundation';
+    return isFoundation ? 'alert-box ' : 'alert alert-';
+  }
+
+  @computed('flash.type', 'messageStylePrefix', '_defaultMessageStylePrefix')
   get alertType() {
     const flashType = this.flash.type || '';
-    const messageStyle = this.messageStyle || '';
-    let prefix = 'alert alert-';
-
-    if (messageStyle === 'foundation') {
-      prefix = 'alert-box ';
-    }
-
+    const prefix = this.messageStylePrefix || this._defaultMessageStylePrefix || '';
```

Comment:
Is there a reason we have `|| ''`? Isn't it impossible for `_defaultMessageStylePrefix` to return a falsy value?

Reply:
You're right, was just being overly careful